ClosedXML_Tests/Excel/CalcEngine/TextTests.cs

```diff
@@ -489,6 +489,52 @@ public void Text_String_Input()
+        [TestCase("=TEXTJOIN(0, FALSE, A1:B2)", "A00B0D")]
+        [TestCase("=TEXTJOIN(false, FALSE, A1:B2)", "AFALSEFALSEBFALSED")]
+        [TestCase("=TEXTJOIN(\",\", 0, A1:B2)", "A,,B,D")]
+        [TestCase("=TEXTJOIN(\",\", 100, A1:B2)", "A,B,D")]
+        [TestCase("=TEXTJOIN(B2, FALSE, A1:B2)", "ADDBDD")]
+        [TestCase("=TEXTJOIN(\",\", FALSE, 12345.67, DATE(2018, 10, 30))", "12345.67,43403")]
+        public void TextJoin(string formula, string expectedOutput)
+        {
+            var wb = new XLWorkbook();
+            IXLWorksheet ws = wb.AddWorksheet("Sheet1");
+            ws.Cell("A1").Value = "A";
+            ws.Cell("A2").Value = "B";
+            ws.Cell("B1").Value = "";
+            ws.Cell("B2").Value = "D";
+
+            ws.Cell("C1").FormulaA1 = formula;
+            var a = ws.Cell("C1").Value;
+
+            Assert.AreEqual(expectedOutput, a.ToString());
+        }
+
+        [TestCase("=TEXTJOIN(\",\", FALSE, D1:D32769)", "The value is too long")]
+        [TestCase("=TEXTJOIN(\",\", \"0\", A1:B2)", "The second argument is invalid")]
```

Comment:
Is this what Excel does for this case? I have Excel 2013, so can't test it. I would actually prefer if we could cast the `"0"` to an actual `bool` of value `false`. That makes more sense to me.

Reply:
I suggest we accept strings that are parsable to `bool` there too. Of course `TEXTJOIN(",", "asdf", A1:B2)` should throw a `CellValueException`.